### ISSUE TITLE
Do we have any plans to support national cloud such as mooncake?

### ISSUE THREAD

**Issue by pjshi23** [projectkudu/ARMExplorer#116]
Do we have any plans to support national cloud such as mooncake?

ARM mode is ready on mooncake so i'm curious do we have any plans for that.

**Comment by davidebbo** [comment]
No concrete plan, though it certainly would be nice.

Unfortunately, this project only has limited funding at this point.

**Comment by gitralf** [comment]
I was about to ask the same question for the german cloud...

**Comment by davidebbo** [comment]
One 'cheap and dirty' approach I discussed with @shobak101 and @suwatch:

- We could host an instance of Explorer that does not itself deal with authentication
- The user would be prompted to choose their national cloud (MC, German, ...) so we know what ARM endpoints to use.
- User would also be prompted to paste their AAD token, which they could obtain using F12 tools on a request to the National Cloud portal. This is definitely a dirty/hacky step!

Not a perfect solution, but potential a stopgap until the pieces are in place that would allow full integration (quite a bit more involved).

**Comment by zakramer** [comment]
AAD is supposed to have a discovery API that will tell you want cloud a tenant is coming from.  We can check and see if that is ready in the long term.

Another thought is just to make it a config switch and deploy an instance for each sovereign?  Not sure if that is easier to dev but harder to maintain.

**Comment by gitralf** [comment]
at least for germany you might consider to use a new URL since there are some restrictions in place, so you cannot use the current subscription for ARMExplorer anyways. You would have to setup a new deployment for Germany and maybe use the URL resources.microsoftazure.de (aligning to the portal URL for Germany). That would solve the need to distinguish which AAD to use since - as now - there is only one available.

**Comment by davidebbo** [comment]
Yes, we'll most likely end up with a different deployment in each cloud.

**Comment by pburkholder** [comment]
+1 to usgov support

**Comment by suwatch** [comment]
@davidebbo The support for copy/paste token would be useful and could be beneficial for other things (such as testing service principal).   We could discover the env from token - so no env selection needed. 

From browser, the OAuth token (not id_token) is encrypted and stored in the session cookie.   Hence, F12 to copy token will not work.  Currently one would need to use ARMClient to achieve that which I think it is an OK workaround till we have a full fetch support on National cloud.

**Comment by davidebbo** [comment]
@suwatch understood. Main problem is finding resources to do this.

**Comment by anirudhcavale** [comment]
+1 to having support for this in National Clouds. Hope you the resourcing issue gets addressed soon @davidebbo

**Comment by mcrosstx** [comment]
Has there been any traction on getting this pushed out into the Government space? I love using ARE in commercial, but now that I'm working with more Government tenants I'm really missing the functionality.

**Comment by gsacavdm** [comment]
Out of curiosity, is there something missing in the Resource Explorer in the Azure portal that's leading you to use this separate version of Resource Explorer?

![resource-explorer](https://user-images.githubusercontent.com/3582236/38688045-200546ee-3e46-11e8-816f-67faea4d598d.png)

**Comment by fume** [comment]
Still no support for National Clouds? Any roadmap?
I understand that the problem are resources: @davidebbo , can we somehow give visibility of this feature request and help you get resources to accomplish this?
thanks, regards

**Comment by abatishchev** [comment]
+1 for Germany cloud. Doesn't seem the Resources portal exists there.
